File: VB/HowToUseTreeListEditor.Module/ProjectArea.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base.General
Imports System.ComponentModel

Namespace HowToUseTreeListEditor.Module

    Public Class ProjectArea
        Inherits Category

        Private projectField As Project

        Protected Overrides ReadOnly Property ParentProp As ITreeNode
            Get
                Return projectField
            End Get
        End Property

        Protected Overrides ReadOnly Property ChildrenProp As IBindingList
            Get
                Return New BindingList(Of Object)()
            End Get
        End Property

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Sub New(ByVal session As Session, ByVal name As String)
            MyBase.New(session)
            NameProp = name
        End Sub

        <Association("Project-ProjectAreas")>
        Public Property Project As Project
            Get
                Return projectField
            End Get

            Set(ByVal value As Project)
                SetPropertyValue("Project", projectField, value)
            End Set
        End Property
    End Class
End Namespace

File: CS/XPO/UseTreeList/UseTreeList.Module/BusinessObjects/ProjectArea.cs
using DevExpress.Xpo;
using DevExpress.Persistent.Base.General;
using System.ComponentModel;

namespace HowToUseTreeListEditor.Module {
    public class ProjectArea : Category {
        private Project project;
        protected override ITreeNode GetParent() {
            return project;
        }
        protected override IBindingList GetChildren() {
            return new BindingList<object>();
        }
        public ProjectArea(Session session) : base(session) { }
        public ProjectArea(Session session, string name)
            : base(session) {
            this.Name = name;
        }
        [Association("Project-ProjectAreas")]
        public Project Project {
            get {
                return project;
            }
            set {
                SetPropertyValue("Project", ref project, value);
            }
        }
    }
}
